02-peer-protocol.md

```diff
@@ -1437,6 +1440,206 @@ a large feerate change, instead sets their `sats` to zero, and decline to
+[quiescent](#channel-quiescence).
+
+Operation returns to normal once negotiation is done (while waiting
+for the splice transaction(s) to confirm).
+
+The splice is finally terminated when both sides send
+`splice_locked` to indicate that one of the splice transactions
+reached acceptable depth.
+
+### The `splice` Message
+
+1. type: 80 (`splice`)
+2. data:
+   * [`channel_id`:`channel_id`]
+   * [`chain_hash`:`chain_hash`]
+   * [`s64`:`relative_amount`]
+   * [`u32`:`funding_feerate_perkw`]
+   * [`u32`:`locktime`]
+   * [`point`:`funding_pubkey`]
+
+1. type: 81 (`splice_ack`)
+2. data:
+   * [`channel_id`:`channel_id`]
+   * [`chain_hash`:`chain_hash`]
+   * [`s64`:`relative_amount`]
+   * [`point`:`funding_pubkey`]
+
+`funding_satoshis` is the amount the sender is putting into the
+post-splice channel. It includes their old funded channel amount
+plus any satoshis they intend to add, or, less any satoshis they
+intend to remove.
+
+#### Requirements
+
+The sender of `splice`:
+- MUST have successfully initiated quiescence.
+- MUST NOT send `splice` before sending and receiving `channel_ready`.
+- MUST NOT send `splice` while a splice is being negotiated.
+- If any splice is in progress:
+  - MUST NOT send a splice message with `funding_feerate_perkw` which is less than 1.25 the previous `funding_feerate_perkw` (rounded down).
+
+The receiver of `splice`:
+- SHOULD fail the connection if there is an pending splice, and the `funding_feerate_perkw` is not at least 1.25 the previous `funding_feerate_perkw` (rounded down).
+- MUST respond with `splice_ack` containing its own `funding_pubkey`.
+- MUST begin splice negotiation.
+
+The receiver of `splice_ack`:
+- MUST begin splice negotiation.
+
+
+#### Rationale
+
+There is no harm in agreeing to a splice with a high feerate
+(presumably the recipient will not contribute to the splice which they
+consider to be overpaying).
+
+## Splice Negotiation
+
+The splice negotiation is very similar to the `tx_init_rbf` negotiation:
+both sides alternate sending `tx_add_input` and `tx_add_output` until
+they both send consecutive `tx_complete`.
+
+### Requirements
+
+(Note that many of these messages have their own, additional
+requirements detailed elsewhere).
+
+The initiator:
+- MUST `tx_add_input` an input which spends the current funding transaction output.
+- MUST `tx_add_output` a zero-value output which pays to the two funding keys using the higher of the two `generation` fields.
+- MUST pay for the common fields.
+
+Each node:
+- MUST pay for their own added inputs and outputs.
+
+Upon receipt of consecutive `tx_complete`s, each node:
+- MUST fail negotiation if there is not exactly one input spending the current funding transaction.
+- MUST fail negotiation if there is not exactly one output with zero value paying to the two funding keys (a.k.a. the new channel funding output)
+- MUST calculate the channel capacity for each side:
```

Comment:
That's not how it works anymore since we started using a signed funding contribution (that is a delta of the what each peer is adding or removing), that section should be re-worked.